src/WarcraftLegacies.Source/Quests/Dalaran/QuestGreymaneWall.cs

```diff
@@ -0,0 +1,76 @@
+﻿using System.Collections.Generic;
+using System.IO;
+using MacroTools.ControlPointSystem;
+using MacroTools.Extensions;
+using MacroTools.FactionSystem;
+using MacroTools.ObjectiveSystem.Objectives.ControlPointBased;
+using MacroTools.ObjectiveSystem.Objectives.FactionBased;
+using MacroTools.ObjectiveSystem.Objectives.QuestBased;
+using MacroTools.ObjectiveSystem.Objectives.TimeBased;
+using MacroTools.ObjectiveSystem.Objectives.UnitBased;
+using MacroTools.QuestSystem;
+using WCSharp.Shared.Data;
+using static War3Api.Common;
+
+namespace WarcraftLegacies.Source.Quests.Dalaran
+{
+  /// <summary>
+  /// Dalaran unlocks and takes control of Gilneas.
+  /// </summary>
+  public sealed class QuestGreymaneWall : QuestData
+  {
+    private readonly unit _gilneasDoor;
+    private readonly ObjectiveAnyUnitInRect _enterGilneasGateRegion;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="QuestGreymaneWall"/> class.
+    /// </summary>
+    /// /// <param name="gilneasDoor">This unit will be transferred to the completeing player.</param>
+    public QuestGreymaneWall(unit gilneasDoor) : base("Greymane Wall",
+      "The Gilneans, fearful of a potential invasion from the frozen north, sealed themselves behind the Greymane Wall. If we are to survive the coming storm, we must force our neighbour back out into the open.",
+      @"ReplaceableTextures\CommandButtons\BTNGate.blp")
+    {
+      _gilneasDoor = gilneasDoor;
+      AddObjective(new ObjectiveControlPoint(ControlPointManager.Instance.GetFromUnitType(Constants.UNIT_N01D_SILVERPINE_FOREST)));
```

Comment:
Rather than copying the quest requirements from QuestDalaran, we should use the `ObjectiveQuestComplete` objective. You'll need to pass the instance of `QuestDalaran` into this quest to do that. You can find a decent example of that here: https://github.com/AzerothWarsLR/WarcraftLegacies/blob/dev/src/WarcraftLegacies.Source/Quests/Lordaeron/QuestCapitalCity.cs

It'll be easier to maintain and will be a clearer UX.